hw-model/src/lib.rs

```diff
@@ -216,4 +281,46 @@ mod tests {
             "expected output \"ha\", was \"hi\""
         );
     }
+
+    #[test]
+    pub fn test_upload_firmware() {
+        let firmware: Vec<u8> = [
```

Comment:
Can you add a comment describing what this firmware does?

Reply:
Oh wait, this doesn't appear to be used

Reply:
This is a test routine for the upload_firmware method. I can add the comment.

Reply:
Oh, sorry, this is separate from the ROM used above, I'm crazy.